saleor/checkout/migrations/0056_move_checkout_metadata_to_separate_model.py

```diff
@@ -0,0 +1,64 @@
+from django.db import migrations
+from django.db.models import Q
+
+
+BATCH_SIZE = 10000
+
+
+def queryset_in_batches(queryset):
+    """Slice a queryset into batches.
+
+    Input queryset should be sorted by pk.
+    """
+    start_pk = 0
+
+    while True:
+        qs = queryset.filter(pk__gt=start_pk)[:BATCH_SIZE]
+        pks = list(qs.values_list("pk", flat=True))
+
+        if not pks:
+            break
+
+        yield pks
+
+        start_pk = pks[-1]
+
+
+def clear_checkout_metadata(checkout):
+    checkout.metadata = None
+    checkout.private_metadata = None
+
+
+def move_all_checkout_metadata(apps, schema_editor):
+    Checkout = apps.get_model("checkout", "Checkout")
+    CheckoutMetadata = apps.get_model("checkout", "CheckoutMetadata")
+    checkouts_with_meta = (
+        Checkout.objects.filter(
+            Q(metadata__isnull=False) | Q(private_metadata__isnull=False)
+        )
+        .order_by("pk")
+        .distinct("token")
+    )
+    for batch_pks in queryset_in_batches(checkouts_with_meta):
+        checkouts = Checkout.objects.filter(pk__in=batch_pks)
+        CheckoutMetadata.objects.bulk_create(
```

Comment:
@saleor/sre 
Could we use `bulk_create` without `queryset_in_batches`?

Reply:
I can't seem to find a way since we need a checkout PK. I thought maybe we could loop it as a subquery to return the PKs to outer func but that would nohow work with bulk_create later on. I think it should be fine how it is, there is just a memory thing to improve but I will post it as a separate comment.